SocketRocket/SRWebSocket.m

```diff
@@ -644,16 +644,22 @@ - (BOOL)sendString:(NSString *)string error:(NSError **)error
 
 - (BOOL)sendData:(nullable NSData *)data error:(NSError **)error
 {
+    data = [data copy];
+    return [self sendDataNoCopy:data error:error];
+}
+
+- (BOOL)sendDataNoCopy:(nullable NSData *)data error:(NSError **)error
+{
     if (self.readyState != SR_OPEN) {
-        NSString *message = @"Invalid State: Cannot call `sendData:error:` until connection is open.";
+        NSString *message = @"Invalid State: Cannot call `sendDataNoCopy:error:` until connection is open.";
         if (error) {
             *error = SRErrorWithCodeDescription(2134, message);
         }
         SRDebugLog(message);
         return NO;
     }
 
-    data = [data copy];
+    NSAssert(self.readyState != SR_CONNECTING, @"Invalid State: Cannot call `sendDataNoCopy:error:` until connection is open");
```

Comment:
Probably this is no longer needed, since we have converted to an error pointer model.